src/spectral.ts

```diff
@@ -23,13 +25,13 @@ export * from './types';
 export class Spectral {
   private _rules: RuleCollection = {};
   private _functions: FunctionCollection = defaultFunctions;
+  private _resolver: Resolver;
 
-  private resolver: Resolver;
-  constructor(opts?: IConstructorOpts) {
-    this.resolver = opts && opts.resolver ? opts.resolver : new Resolver();
+  constructor(opts: IConstructorOpts = {}) {
+    this._resolver = opts.resolver ? opts.resolver : new Resolver();
   }
 
-  public async run(target: IParsedResult | object | string): Promise<IRuleResult[]> {
+  public async run(target: IParsedResult | object | string, opts: IRunOpts = {}): Promise<IRuleResult[]> {
```

Comment:
This looks a bit weird. I believe we should let resolver be configurable rather than providing its settings to `run` method.
Thoughts?

Reply:
The `authority` is not a global resolver option, since it kind of depends on the file being run, so that option won't really work.